src/server.rs

```diff
@@ -135,7 +137,7 @@ impl Server {
         }
     }
 
-    pub async fn start(&mut self) {
+    pub async fn start(&mut self, rx: Option<mpsc::Receiver<()>>) {
```

Comment:
I understand this PR is still in Draft, but why do you wanna introduce a channel over here ?

Reply:
Hey 👋🏻 
No worries, feel free to review anytime I'm completely fine with it. 

This is added because if I don't pass some sort of signal, the test case will keep running forever because of the infinite `loop`.  I couldn't think of any better approach than this that's why I consider it optional so actual implementation/usage won't have to pass any value. I'm planning to add it in other methods too where we have an infinite `loop`.

I'm happy to learn some new techniques/patterns if you can share them here.

Thanks!